add vendors.md doc about what it means to support opentelemetry (#594)
open-telemetry/opentelemetry-specification: specification/vendors.md

</details>

## Abstract

The OpenTelemetry project consists of both a
[specification](https://github.com/open-telemetry/opentelemetry-specification)
for the API, SDK, protocol and semantic conventions, as well as an
implementation of each for a number of languages. The default SDK implementation
is [highly configurable](sdk-configuration.md) and extendable, for example
through [Span Processors](trace/sdk.md#span-processor), to allow for additional
logic needed by particular vendors to be added without having to implement a
custom SDK. By not requiring a custom SDK means for most languages a user will
already find an implementation to use and if not they'll have a well documented
specification to follow for implementing in a new language.

The goal is for users to be able to easily switch between vendors while also
ensuring that any language with an OpenTelemetry SDK implementation is able to
work with any vendor who claims support for OpenTelemetry.

This document will explain what is required of a vendor to be considered to
"Support OpenTelemetry" or "Implements OpenTelemetry".

## What it means to support OpenTelemetry

> ## Tracing, Metrics and Logging Qualifications

Reviewer: Probably having the sections instead as `Implements OpenTelemetry` and `Supports OpenTelemetry` (with their respective clarifications and side notes) will be easier for new users to grasp.

Author: Good point, I'll update them.

## Abstract

The OpenTelemetry project consists of both a
[specification](https://github.com/open-telemetry/opentelemetry-specification)
for the API, SDK, protocol and semantic conventions, as well as an
implementation of each for a number of languages. The default SDK implementation
is [highly configurable](sdk-configuration.md) and extendable, for example
through [Span Processors](trace/sdk.md#span-processor), to allow for additional
logic needed by particular vendors to be added without having to implement a
custom SDK. By not requiring a custom SDK means for most languages a user will
already find an implementation to use and if not they'll have a well documented
specification to follow for implementing in a new language.

The goal is for users to be able to easily switch between vendors while also
ensuring that any language with an OpenTelemetry SDK implementation is able to
work with any vendor who claims support for OpenTelemetry.

This document will explain what is required of a vendor to be considered to
"Support OpenTelemetry" or "Implements OpenTelemetry".

## Supports OpenTelemetry

"Supports OpenTelemetry" means the vendor must accept the output of the default

> SDK through the OpenTelemetry Protocol either directly -- meaning the vendor's

Reviewer 2: ```suggestion
SDK through one of two mechanisms:

- By providing an exporter for the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/) and / or the OpenTelemetry SDKs
- By building a receiver for the OpenTelemetry protocol

```

> endpoint/collector/agent implements the receiver end of the protocol -- or as an

Reviewer 2: ```suggestion

```

Author: I think merging your first suggestion has messed up the others my shifting the lines and columns. I think I know what they are meant to be changing so will do locally the changes and push them.

> SDK through one of two mechanisms:
> 
> - By providing an exporter for the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/) and / or the OpenTelemetry SDKs

Reviewer 3: ```suggestion
SDK through one of the following mechanisms:

- By providing an exporter for the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/) and / or the OpenTelemetry SDKs
- By providing an exporter for the OpenTelemetry SDKs they support
```

I would argue a vendor wouldn't have to implement exporters for *all* language implementations out there but only for the ones they claim support for.

Author: I would not consider that "supporting opentelemetry". If we want to add an option that explicitly limits what their support is that is fine, but the goal here is to not allow vendors to claim "support for opentelemetry" and for users to then find out the otel sdk they use isn't supported.

Reviewer 3: I see. I think it would make sense to add the (SDK) language as a qualification in addition to the signal types (traces|metrics|logs) then.

> exporter that works with the [OpenTelemetry

Reviewer 2: ```suggestion
```

> - By building a receiver for the OpenTelemetry protocol

Reviewer 3: ```suggestion
- By building a receiver for the [OpenTelemetry protocol](https://github.com/open-telemetry/opentelemetry-proto)
```

> Collector](https://github.com/open-telemetry/opentelemetry-collector/).

Reviewer 2: ```suggestion
```
